### ISSUE TITLE
Could you provide a JAR that can be placed into the mod folder or instructions on how to create one?

### ISSUE THREAD

**Issue by HileyNoteson** [BurnhamR/McHeli2#7]
I just want to try it in its current state but I cannot for the life of me figure out how to compile the source

Many Thanks
HileyNotesosn

**Comment by RamennBowls** [comment]
Inside of the MCHeli2 folder, you can import the gradle project into an IDE like Eclipse or Intelij and compile the mod but I've heard that they were doing most things in a private repo so I don't know how much is inside of this public repo.

**Comment by RamennBowls** [comment]
There is actually a README file inside of the repo that explains how to get the project setup but I personally prefer to do it a different way, but it is there encase you get stuck.

**Comment by RamennBowls** [comment]
They were working on everything in a private repo some of which has been made public here, I'm not sure if the project has been abandoned though but I know that this public repo isn't up-to-date...

I use the Intelij IDE, it might be relatively the same if you are using Eclipse but I'm unsure.

Intelij Instructions:
1. You'll want to download the repo and extract the McHeli2 folder.
2. To import that gradle project, you can just go to File -> Open and open the `build.gradle` file and it'll import the gradle project.
3. Once the gradle project is open, make sure you have the Gradle window (if you don't have it, go to View -> Tool Windows -> Gradle).
4. To compile it, inside of the Gradle Window go to the build folder and run build. The compiled mod will saved in your project folder build -> libs.

If you want to modify any code, inside of the Gradle Window first run `setupDecompWorkspace` inside of the forgegradle folder and then reload the Gradle Project.

**Comment by HileyNoteson** [comment]
> 
> 
> They were working on everything in a private repo some of which has been made public here, I'm not sure if the project has been abandoned though but I know that this public repo isn't up-to-date...
> 
> I use the Intelij IDE, it might be relatively the same if you are using Eclipse but I'm unsure.
> 
> Intelij Instructions:
> 
>     1. You'll want to download the repo and extract the McHeli2 folder.
> 
>     2. To import that gradle project, you can just go to File -> Open and open the `build.gradle` file and it'll import the gradle project.
> 
>     3. Once the gradle project is open, make sure you have the Gradle window (if you don't have it, go to View -> Tool Windows -> Gradle).
> 
>     4. To compile it, inside of the Gradle Window go to the build folder and run build. The compiled mod will saved in your project folder build -> libs.
> 
> 
> If you want to modify any code, inside of the Gradle Window first run `setupDecompWorkspace` inside of the forgegradle folder and then reload the Gradle Project.

this is what i get when I build it

15:19:22: Executing task 'build'...

This mapping 'snapshot_20171003' was designed for MC 1.12! Use at your own peril.
WARNING: You are using an unsupported version of ForgeGradle.
Please consider upgrading to ForgeGradle 4 and helping in the efforts to get old versions working on the modern toolchain.
See https://gist.github.com/TheCurle/fe7ad3ede188cbdd15c235cc75d52d4a for more info on contributing.
:deobfCompileDummyTask
:getVersionJson
:extractUserdev
:downloadClient SKIPPED
:downloadServer SKIPPED
:splitServerJar SKIPPED
:mergeJars SKIPPED
:applyBinaryPatches SKIPPED
:deobfProvidedDummyTask
:extractDependencyATs SKIPPED
:extractMcpData SKIPPED
:extractMcpMappings SKIPPED
:genSrgs SKIPPED
:deobfMcMCP SKIPPED
:sourceApiJava
:compileApiJava UP-TO-DATE
:processApiResources UP-TO-DATE
:apiClasses UP-TO-DATE
:sourceMainJava
:compileJavaC:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\test.java:9: error: class MCH_CommonProxy is public, should be declared in a file named MCH_CommonProxy.java
/*    */ public class MCH_CommonProxy {
                ^
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\MCH_MOD.java:19: error: cannot find symbol
	public static CommonProxy proxy;
	              ^
  symbol:   class CommonProxy
  location: class MCH_MOD
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:4: error: package cpw.mods.fml.common does not exist
import cpw.mods.fml.common.FMLCommonHandler;
                          ^
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:39: error: cannot find symbol
	public String side() { return "Server"; } public MCH_SoundUpdater CreateSoundUpdater(MCH_EntityAircraft aircraft) {
	                                                                                     ^
  symbol:   class MCH_EntityAircraft
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:39: error: cannot find symbol
	public String side() { return "Server"; } public MCH_SoundUpdater CreateSoundUpdater(MCH_EntityAircraft aircraft) {
	                                                 ^
  symbol:   class MCH_SoundUpdater
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:45: error: cannot find symbol
	public MCH_Config loadConfig(String fileName) {
	       ^
  symbol:   class MCH_Config
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:53: error: cannot find symbol
	public MCH_Config reconfig() 
	       ^
  symbol:   class MCH_Config
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:86: error: cannot find symbol
	public void printChatMessage(IChatComponent chat, int showTime, int pos) {}
	                             ^
  symbol:   class IChatComponent
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\test.java:3: error: package cpw.mods.fml.common does not exist
/*    */ import cpw.mods.fml.common.FMLCommonHandler;
                                   ^
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\test.java:4: error: package mcheli.aircraft does not exist
/*    */ import mcheli.aircraft.MCH_SoundUpdater;
                               ^
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\test.java:7: error: cannot find symbol
/*    */ import net.minecraft.util.IChatComponent;
                                  ^
  symbol:   class IChatComponent
  location: package net.minecraft.util
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\MCH_MOD.java:18: error: cannot find symbol
	@SidedProxy(clientSide = Reference.CLIENT_PROXY_CLASS, serverSide = REFERENCE.COMMON_PROXY_CLASS)
	                                                                    ^
  symbol:   variable REFERENCE
  location: class MCH_MOD
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:22: error: cannot find symbol
		return MinecraftServer.func_71276_C().func_71270_I();
		                      ^
  symbol:   method func_71276_C()
  location: class MinecraftServer
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:35: error: cannot find symbol
		FMLCommonHandler.instance().bus().register(new MCH_ServerTickHandler());
		                                               ^
  symbol:   class MCH_ServerTickHandler
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:47: error: cannot find symbol
		MCH_Config config = new MCH_Config("./", fileName);
		^
  symbol:   class MCH_Config
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:47: error: cannot find symbol
		MCH_Config config = new MCH_Config("./", fileName);
		                        ^
  symbol:   class MCH_Config
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:55: error: cannot find symbol
		MCH_Lib.DbgLog(false, "MCH_CommonProxy.reconfig()", new Object[0]);
		^
  symbol:   variable MCH_Lib
  location: class MCH_CommonProxy
C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java:81: error: cannot find symbol
		return MinecraftServer.func_71276_C().func_71264_H();
		                      ^
  symbol:   method func_71276_C()
  location: class MinecraftServer
Note: C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\proxy\MCH_CommonProxy.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
18 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
 FAILED

BUILD FAILED

Total time: 11.702 secs
15:19:34: Task execution finished 'build'.

**Comment by RamennBowls** [comment]
Yep, there are multiple errors. What ever the developers coded in, doesn't actually work hence the error.

I'm no java expert but you can see the first error says `error: cannot find symbol
public static CommonProxy proxy;` also saying the error is on line 19 in the MCH_MOD class as shown by `C:\Users\bookl\Downloads\McHeli2\build\sources\main\java\com\skyy\mcheli\MCH_MOD.java:19`.

You'll need to modify the broken code to fix it. On their private repo it probably but considering this public one is out-of-date (last updated 7 months ago), I'm not surprised its broken.

**Comment by HileyNoteson** [comment]
i think the problem is the extra proxy in 19 with the extra proxy, I try to remove it but building/debuging/etc justs reverts it. What am I missing?

**Comment by HileyNoteson** [comment]
wait nvm it was not referencing what I thought. It is referenceing the util file, both of them. Nothing seems wrong, but clearly something is

**Comment by RamennBowls** [comment]
Okay, so I decided to try and fix it so I could compile it, and well... the thing is that its like 1% of the actual mod?
As I was fixing a lot of the errors, I noticed that there we're just a lot of unused classes, a few of the issues being that these classes literally don't exist?

For example the MCH_MOD class uses the MCH_CommonProxy class, and that works fine because it exists, but within the MCH_CommonProxy class its trying to use other classes including MCH_ServerTickHandler and MCH_Config which literally don't exist.

There are also just a LOAD of methods being registered but not being used, I assume that's because this is like 1% of the actual mod. I can fix everything but there's literally nothing in the mod if there is anything.


**TL;DR, there is hardly any code in this public repo, what we have is like 1% of the actual mod, it has code for things, but those classes don't exist in this public repo which explains the errors as its basically 1% of the actual mod. I can fix everything, but with what we have it's just the shell, an empty mod with no content.**
